python_modules/dagster/dagster/_cli/code_server.py

```diff
@@ -144,6 +146,21 @@ def code_server_cli():
     help="How long to wait for code to load or reload before timing out. Defaults to no timeout.",
     envvar="DAGSTER_CODE_SERVER_STARTUP_TIMEOUT",
 )
+@click.option(
+    "--heartbeat",
+    is_flag=True,
+    help=(
+        "If set, the GRPC server will shut itself down when it fails to receive a heartbeat "
+        "after a timeout configurable with --heartbeat-timeout."
+    ),
+)
+@click.option(
+    "--heartbeat-timeout",
+    type=click.INT,
+    required=False,
+    default=DEFAULT_HEARTBEAT_TIMEOUT,
+    help="How long to wait for a heartbeat from the caller before timing out. Defaults to 30 seconds.",
```

Comment:
maybe worth clarifying here that this only comes into play if --heartbeat is set